src/main/java/store/domain/Product.java

```diff
@@ -0,0 +1,136 @@
+package store.domain;
+
+import static store.constants.ProductConstants.OUT_OF_STOCK;
+import static store.constants.ProductConstants.PRODUCT_DESCRIPTION_PREFIX;
+import static store.constants.ProductConstants.UNIT_QUANTITY;
+import static store.constants.ProductConstants.UNIT_WON;
+import static store.message.ErrorMessage.NOT_FOUND_PROMOTION;
+
+import java.text.NumberFormat;
+import java.util.Locale;
+import java.util.Optional;
+import store.dto.ProductDto;
+import store.validation.ProductValidator;
+
+public class Product {
+    private String name;
+    private int price;
+    private int quantity;
+    private Optional<Promotion> promotion;
+
+    private Product() {
+    }
+
+    public Product(ProductDto productDto, Optional<Promotion> promotion) {
+        ProductValidator.validate(productDto);
+        this.name = productDto.name();
+        this.price = Integer.parseInt(productDto.price());
+        this.quantity = Integer.parseInt(productDto.quantity());
+        this.promotion = promotion;
+    }
+
+    /**
+     * 프로모션을 적용 여부 검증
+     */
+    public boolean isEligibleForStandardPromotion(int orderedQuantity) {
+        if (promotion.isEmpty()) {
+            return false;
+        }
+        Promotion promotionInfo = promotion.get();
+        return orderedQuantity <= quantity
+                && getRemainingItemsForPromotion(orderedQuantity, promotionInfo) + promotionInfo.getGet() <= quantity;
+    }
```

Comment:
Optional의 기능을 활용해보면 어떨까요!?
```suggestion
    public boolean isEligibleForStandardPromotion(int orderedQuantity) {
        return promotion.map(promotionInfo -> 
            orderedQuantity <= quantity &&
            getRemainingItemsForPromotion(orderedQuantity, promotionInfo) + promotionInfo.getGet() <= quantity
        ).orElse(false);
    }
```